assets/robotest/Makefile

```diff
@@ -0,0 +1,306 @@
+ROBOTEST_GOOGLE_APPLICATION_CREDENTIALS ?= $(TOP)/.gcp-credentials.json
+
+# kudos to https://gist.github.com/prwhite/8168133 for inspiration
+.PHONY: help
+help: ## Show this message.
+	@echo 'Usage: make [options] [target] ...'
+	@echo
+	@echo 'Options: run `make --help` for options'
+	@echo
+	@echo 'Targets:'
+	@egrep --no-filename '^(.+)\:\ ##\ (.+)' ${MAKEFILE_LIST} | column -t -c 2 -s ':#' | sort | sed 's/^/  /'
+
+.PHONY: clean
+clean: ## Remove build artifacts.
+	rm -rf $(ROBOTEST_BUILDDIR)
+
+.PHONY: clean-caches
+clean-caches: ## Remove cached tele binaries and gravity packages.
+	rm -rf $(ROBOTEST_CACHES) $(ROBOTEST_CACHE_TMP)
+
+$(ROBOTEST_BUILDDIR):
+	mkdir -p $(ROBOTEST_BUILDDIR)
+
+$(ROBOTEST_IMAGEDIR):
+	mkdir -p $(ROBOTEST_IMAGEDIR)
+
+$(ROBOTEST_TELE_CACHE):
+	mkdir -p $(ROBOTEST_TELE_CACHE)
+
+$(ROBOTEST_GRAVITY_PKG_CACHE):
+	mkdir -p $(ROBOTEST_GRAVITY_PKG_CACHE)
+
+$(ROBOTEST_BUILD_TMP):
+	mkdir -p $(ROBOTEST_BUILD_TMP)
+
+$(ROBOTEST_CACHE_TMP):
+	mkdir -p $(ROBOTEST_CACHE_TMP)
+
+.PHONY: images
+images: ## Build all images necessary to run robotest CI tests.
+images: image-telekube image-opscenter image-robotest image-robotest-upgrade
+
+.PHONY: image-telekube
+image-telekube: ## Move telekube.tar cluster image into place for test execution.
+image-telekube: $(ROBOTEST_TELEKUBE_IMAGE)
+
+$(ROBOTEST_TELEKUBE_IMAGE): $(TELEKUBE_OUT) | $(ROBOTEST_IMAGEDIR)
+	cp $(TELEKUBE_OUT) $(ROBOTEST_TELEKUBE_IMAGE)
+
+.PHONY: image-opscenter
+image-opscenter: ## Move opscenter.tar cluster image into place for test execution.
+image-opscenter: $(ROBOTEST_OPSCENTER_IMAGE)
+
+$(ROBOTEST_OPSCENTER_IMAGE): $(OPSCENTER_OUT) | $(ROBOTEST_IMAGEDIR)
+	cp $(OPSCENTER_OUT) $(ROBOTEST_OPSCENTER_IMAGE)
+
+.PHONY: image-robotest
+image-robotest: ## Build robotest cluster image using the current tele.
+image-robotest: $(ROBOTEST_IMAGE)
+
+$(ROBOTEST_IMAGE): export TARGET = $(ROBOTEST_IMAGE)
+$(ROBOTEST_IMAGE): export IMAGE = $(ROBOTEST_TELE_BUILD_DOCKER_IMAGE)
+$(ROBOTEST_IMAGE): export BUILD_TMP = $(ROBOTEST_BUILD_TMP)
+$(ROBOTEST_IMAGE): export APP_MANIFEST = $(ROBOTEST_IMAGE_MANIFEST)
+$(ROBOTEST_IMAGE): export APP_SRCDIR = $(ROBOTEST_IMAGE_SRCDIR)
+$(ROBOTEST_IMAGE): export VERSION = $(GRAVITY_VERSION)
+$(ROBOTEST_IMAGE): export TELE = $(TELE_OUT)
+$(ROBOTEST_IMAGE): export STATE_DIR = $(PACKAGES_DIR)
+$(ROBOTEST_IMAGE): $(TELE_OUT) $(ROBOTEST_IMAGE_SRC) $(TOP)/tele_build.sh | $(ROBOTEST_IMAGEDIR) $(ROBOTEST_BUILD_TMP)
+	$(TOP)/tele_build.sh
+
+# When using a shared state dir 7.0.x package syncing is prone to races
+# resulting in failures like:
+#
+#   [ERROR]: failed to sync packages for runtime version 7.0.16
+#       package(gravitational.io/bandwagon:6.0.1) already exists
+#
+# Possibly due to https://github.com/gravitational/gravity/issues/2060
+#
+# Because we must support old tele binaries to build old image versions, these races seem
+# unavoidable.
+#
+# Interestingly, I've never seen sync races happen with oss tele and the s3Syncer
```

Comment:
This is b/c oss syncer implementation always downloads into an abritrary temporary directory.